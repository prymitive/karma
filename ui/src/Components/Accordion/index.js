import React from "react";

import Collapsible from "react-collapsible";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronUp } from "@fortawesome/free-solid-svg-icons/faChevronUp";
import { faChevronDown } from "@fortawesome/free-solid-svg-icons/faChevronDown";

const Trigger = ({ text, isOpen }) => (
  <div className="d-flex flex-row justify-content-between">
    <div>{text}</div>
    <div>
      <FontAwesomeIcon
        icon={isOpen ? faChevronDown : faChevronUp}
        className="text-muted"
      />
    </div>
  </div>
);

const Accordion = ({ text, content, extraProps }) => (
  <Collapsible
    triggerTagName="div"
    transitionTime={50}
    className="card"
    openedClassName="card"
    triggerClassName="card-header cursor-pointer"
    triggerOpenedClassName="card-header cursor-pointer bg-light"
    contentOuterClassName="collapse show"
    contentInnerClassName="card-body my-2"
    {...extraProps}
    trigger={<Trigger text={text} isOpen={false} />}
    triggerWhenOpen={<Trigger text={text} isOpen={true} />}
  >
    {content}
  </Collapsible>
);

export { Accordion };
