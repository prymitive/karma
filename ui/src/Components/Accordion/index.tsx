import { FC, ReactNode, useState } from "react";

export const AccordionItem: FC<{
  text: string;
  content: ReactNode;
  defaultIsOpen?: boolean;
}> = ({ text, content, defaultIsOpen }) => {
  const [isOpen, setIsOpen] = useState<boolean>(defaultIsOpen || false);

  return (
    <div className="accordion-item">
      <h2 className="accordion-header">
        <button
          className={`accordion-button ${isOpen ? "" : "collapsed"}`}
          type="button"
          onClick={() => setIsOpen((val) => !val)}
        >
          {text}
        </button>
      </h2>
      <div className={`accordion-collapse ${isOpen ? "show" : ""}`}>
        {isOpen ? <div className="accordion-body">{content}</div> : null}
      </div>
    </div>
  );
};

export const Accordion: FC = ({ children }) => {
  return <div className="accordion">{children}</div>;
};
