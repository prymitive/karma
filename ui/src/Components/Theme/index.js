import React, { useEffect } from "react";
import ReactDOM from "react-dom";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSun } from "@fortawesome/free-solid-svg-icons/faSun";

const DarkTheme = React.lazy(() => import("Styles/DarkTheme"));
const LightTheme = React.lazy(() => import("Styles/LightTheme"));

const Placeholder = () => {
  return ReactDOM.createPortal(
    <div
      style={{
        zIndex: 2000,
        backgroundColor: "#455a64",
        position: "fixed",
        top: 0,
        left: 0,
        width: "100vw",
        height: "100vh",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
      }}
    >
      <FontAwesomeIcon
        icon={faSun}
        size="lg"
        spin
        style={{
          color: "#5e7a88",
          fontSize: "14rem",
        }}
      />
    </div>,
    document.body
  );
};

const ThemeContext = React.createContext();

const BodyTheme = () => {
  const context = React.useContext(ThemeContext);

  useEffect(() => {
    document.body.classList.toggle("theme-dark", context.isDark);
    document.body.classList.toggle("theme-light", !context.isDark);
  }, [context.isDark]);

  return (
    <React.Suspense fallback={<Placeholder />}>
      {context.isDark ? <DarkTheme /> : <LightTheme />}
    </React.Suspense>
  );
};

export { BodyTheme, ThemeContext };
