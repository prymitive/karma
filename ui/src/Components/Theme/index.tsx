import React, { FC, useEffect } from "react";
import ReactDOM from "react-dom";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSun } from "@fortawesome/free-solid-svg-icons/faSun";

import { Styles } from "react-select";

const DarkTheme = React.lazy(() => import("Styles/DarkTheme"));
const LightTheme = React.lazy(() => import("Styles/LightTheme"));

const Placeholder = () => {
  return ReactDOM.createPortal(
    <div
      style={{
        zIndex: 2000,
        backgroundColor: "#455a64",
        position: "fixed",
        top: 0,
        left: 0,
        width: "100vw",
        height: "100vh",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
      }}
    >
      <FontAwesomeIcon
        icon={faSun}
        size="lg"
        spin
        style={{
          color: "#5e7a88",
          fontSize: "14rem",
        }}
      />
    </div>,
    document.body
  );
};

export interface ThemeCtx {
  isDark: boolean;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  reactSelectStyles: Styles<any, any>;
  animations: {
    duration: number;
  };
}

const ThemeContext = React.createContext({
  isDark: false,
  reactSelectStyles: {},
  animations: { duration: 1000 },
} as ThemeCtx);

const BodyTheme: FC = () => {
  const context = React.useContext(ThemeContext);

  useEffect(() => {
    document.body.classList.toggle("theme-dark", context.isDark);
    document.body.classList.toggle("theme-light", !context.isDark);
  }, [context.isDark]);

  return (
    <React.Suspense fallback={<Placeholder />}>
      {context.isDark ? <DarkTheme /> : <LightTheme />}
    </React.Suspense>
  );
};

export { BodyTheme, ThemeContext };
