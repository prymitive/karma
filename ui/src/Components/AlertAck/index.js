import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";

import { toJS } from "mobx";
import { useObserver } from "mobx-react-lite";

import addSeconds from "date-fns/addSeconds";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheck } from "@fortawesome/free-solid-svg-icons/faCheck";
import { faCheckCircle } from "@fortawesome/free-solid-svg-icons/faCheckCircle";
import { faSpinner } from "@fortawesome/free-solid-svg-icons/faSpinner";
import { faExclamationCircle } from "@fortawesome/free-solid-svg-icons/faExclamationCircle";

import { APIGroup } from "Models/API";
import { AlertStore } from "Stores/AlertStore";
import {
  SilenceFormStore,
  MatchersFromGroup,
  GenerateAlertmanagerSilenceData,
} from "Stores/SilenceFormStore";
import { useFetchAny } from "Hooks/useFetchAny";
import { TooltipWrapper } from "Components/TooltipWrapper";

const AlertAck = ({ alertStore, silenceFormStore, group }) => {
  const [clusters, setClusters] = useState([]);
  const [upstreams, setUpstreams] = useState([]);
  const [currentCluster, setCurrentCluster] = useState(0);
  const [isAcking, setIsAcking] = useState(false);

  const { response, error, inProgress, reset } = useFetchAny(upstreams);

  const onACK = () => {
    setIsAcking(true);

    let author =
      silenceFormStore.data.author !== ""
        ? toJS(silenceFormStore.data.author)
        : toJS(alertStore.settings.values.alertAcknowledgement.author);

    if (alertStore.info.authentication.enabled) {
      silenceFormStore.data.author = toJS(
        alertStore.info.authentication.username
      );
      author = alertStore.info.authentication.username;
    }

    const alertmanagers = Object.entries(group.alertmanagerCount)
      .filter(([amName, alertCount]) => alertCount > 0)
      .map(([amName, _]) => amName);
    const clusters = Object.entries(
      alertStore.data.clustersWithoutReadOnly
    ).filter(([clusterName, clusterMembers]) =>
      alertmanagers.some((m) => clusterMembers.includes(m))
    );

    let c = [];
    for (const [clusterName, clusterMembers] of clusters) {
      const durationSeconds = toJS(
        alertStore.settings.values.alertAcknowledgement.durationSeconds
      );
      const commentPrefix = toJS(
        alertStore.settings.values.alertAcknowledgement.commentPrefix
      );
      const now = new Date();
      c.push({
        payload: GenerateAlertmanagerSilenceData(
          now,
          addSeconds(now, durationSeconds),
          MatchersFromGroup(group, [], group.alerts, true),
          author,
          `${
            commentPrefix ? commentPrefix + " " : ""
          }This alert was acknowledged using karma on ${now.toUTCString()}`
        ),
        clusterName: clusterName,
        members: clusterMembers,
      });
    }
    setClusters(c);
  };

  useEffect(() => {
    if (upstreams.length && !inProgress && (error || response)) {
      if (clusters.length > currentCluster + 1) {
        setCurrentCluster(currentCluster + 1);
      } else {
        setIsAcking(false);
      }
    }
  }, [clusters, upstreams, currentCluster, inProgress, error, response]);

  useEffect(() => {
    if (clusters.length) {
      reset();
      const cluster = clusters[currentCluster];
      let u = [];
      cluster.members.forEach((amName) => {
        const am = alertStore.data.getAlertmanagerByName(amName);
        if (am !== undefined) {
          u.push({
            uri: `${am.uri}/api/v2/silences`,
            options: {
              method: "POST",
              body: JSON.stringify(cluster.payload),
              credentials: am.corsCredentials,
              headers: {
                "Content-Type": "application/json",
                ...am.headers,
              },
            },
          });
        } else {
          console.error(`Alertmanager "${amName}" not found`);
        }
      });
      setUpstreams(u);
    }
  }, [alertStore.data, clusters, currentCluster, reset]);

  useEffect(() => {
    let timer;
    if (!isAcking && error) {
      timer = setTimeout(() => {
        setUpstreams([]);
        setIsAcking(false);
        reset();
      }, 20 * 1000);
    }
    return () => clearTimeout(timer);
  }, [isAcking, error, reset]);

  return useObserver(() =>
    alertStore.settings.values.alertAcknowledgement.enabled === false ? null : (
      <TooltipWrapper
        title={
          !isAcking && error
            ? error
            : "Acknowledge this alert with a short lived silence"
        }
      >
        <span
          className={`badge badge-pill components-label components-label-with-hover px-2 ${
            !isAcking && error
              ? "badge-warning"
              : !isAcking && response
              ? "badge-success"
              : "badge-secondary"
          }`}
          onClick={() => {
            if (!isAcking && !(response || error)) {
              setIsAcking(true);
              onACK();
            }
          }}
        >
          {!isAcking && error ? (
            <FontAwesomeIcon icon={faExclamationCircle} fixedWidth />
          ) : !isAcking && response ? (
            <FontAwesomeIcon icon={faCheckCircle} fixedWidth />
          ) : isAcking ? (
            <FontAwesomeIcon icon={faSpinner} fixedWidth spin />
          ) : (
            <FontAwesomeIcon icon={faCheck} fixedWidth />
          )}
        </span>
      </TooltipWrapper>
    )
  );
};
AlertAck.propTypes = {
  alertStore: PropTypes.instanceOf(AlertStore).isRequired,
  silenceFormStore: PropTypes.instanceOf(SilenceFormStore).isRequired,
  group: APIGroup.isRequired,
};

export { AlertAck };
