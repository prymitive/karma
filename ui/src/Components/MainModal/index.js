import React, { useState, useCallback } from "react";
import PropTypes from "prop-types";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCog } from "@fortawesome/free-solid-svg-icons/faCog";
import { faSpinner } from "@fortawesome/free-solid-svg-icons/faSpinner";

import { AlertStore } from "Stores/AlertStore";
import { Settings } from "Stores/Settings";
import { TooltipWrapper } from "Components/TooltipWrapper";
import { Modal } from "Components/Modal";

// https://github.com/facebook/react/issues/14603
const MainModalContent = React.lazy(() =>
  import("./MainModalContent").then((module) => ({
    default: module.MainModalContent,
  }))
);

const MainModal = ({ alertStore, settingsStore }) => {
  const [isVisible, setIsVisible] = useState(false);

  const toggle = useCallback(() => setIsVisible(!isVisible), [isVisible]);

  return (
    <React.Fragment>
      <li
        className={`nav-item components-navbar-button ${
          isVisible ? "border-info" : ""
        }`}
      >
        <TooltipWrapper title="Settings">
          <span
            id="components-settings"
            className="nav-link cursor-pointer"
            onClick={toggle}
          >
            <FontAwesomeIcon icon={faCog} />
          </span>
        </TooltipWrapper>
      </li>
      <Modal isOpen={isVisible} toggleOpen={toggle}>
        <React.Suspense
          fallback={
            <h1 className="display-1 text-placeholder p-5 m-auto">
              <FontAwesomeIcon icon={faSpinner} size="lg" spin />
            </h1>
          }
        >
          <MainModalContent
            alertStore={alertStore}
            settingsStore={settingsStore}
            onHide={() => setIsVisible(false)}
            isVisible={isVisible}
            expandAllOptions={false}
          />
        </React.Suspense>
      </Modal>
    </React.Fragment>
  );
};
MainModal.propTypes = {
  alertStore: PropTypes.instanceOf(AlertStore).isRequired,
  settingsStore: PropTypes.instanceOf(Settings).isRequired,
};

export { MainModal };
