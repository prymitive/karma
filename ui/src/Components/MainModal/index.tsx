import React, { FC, useState, useCallback } from "react";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSlidersH } from "@fortawesome/free-solid-svg-icons/faSlidersH";
import { faSpinner } from "@fortawesome/free-solid-svg-icons/faSpinner";

import { AlertStore } from "Stores/AlertStore";
import { Settings } from "Stores/Settings";
import { TooltipWrapper } from "Components/TooltipWrapper";
import { Modal } from "Components/Modal";

// https://github.com/facebook/react/issues/14603
const MainModalContent = React.lazy(() =>
  import("./MainModalContent").then((module) => ({
    default: module.MainModalContent,
  }))
);

const MainModal: FC<{
  alertStore: AlertStore;
  settingsStore: Settings;
}> = ({ alertStore, settingsStore }) => {
  const [isVisible, setIsVisible] = useState<boolean>(false);

  const toggle = useCallback(() => setIsVisible(!isVisible), [isVisible]);

  return (
    <>
      <li
        className={`nav-item components-navbar-button ${
          isVisible ? "border-info" : ""
        } ml-auto`}
      >
        <TooltipWrapper title="Settings">
          <span
            id="components-settings"
            className="nav-link cursor-pointer"
            onClick={toggle}
          >
            <FontAwesomeIcon icon={faSlidersH} fixedWidth />
          </span>
        </TooltipWrapper>
      </li>
      <Modal isOpen={isVisible} toggleOpen={toggle}>
        <React.Suspense
          fallback={
            <h1 className="display-1 text-placeholder p-5 m-auto">
              <FontAwesomeIcon icon={faSpinner} size="lg" spin />
            </h1>
          }
        >
          <MainModalContent
            alertStore={alertStore}
            settingsStore={settingsStore}
            onHide={() => setIsVisible(false)}
            expandAllOptions={false}
          />
        </React.Suspense>
      </Modal>
    </>
  );
};

export { MainModal };
