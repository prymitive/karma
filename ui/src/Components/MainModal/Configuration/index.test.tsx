import { mount } from "enzyme";

import fetchMock from "fetch-mock";

import toDiffableHtml from "diffable-html";

import { MockThemeContext } from "__fixtures__/Theme";
import { Settings } from "Stores/Settings";
import { ThemeContext } from "Components/Theme";
import { Configuration } from ".";

beforeEach(() => {
  fetchMock.reset();
  fetchMock.mock("*", {
    status: 200,
    body: JSON.stringify([]),
  });
});

afterEach(() => {
  jest.restoreAllMocks();
  fetchMock.reset();
});

describe("<Configuration />", () => {
  it("matches snapshot", () => {
    const settingsStore = new Settings(null);
    const tree = mount(
      <ThemeContext.Provider value={MockThemeContext}>
        <Configuration settingsStore={settingsStore} defaultIsOpen={true} />
      </ThemeContext.Provider>
    );
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });
});
