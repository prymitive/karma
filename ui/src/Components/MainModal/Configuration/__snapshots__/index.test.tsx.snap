// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Configuration /> matches snapshot 1`] = `
"
<div class=\\"accordion\\">
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button \\"
              type=\\"button\\"
      >
        Refresh interval
      </button>
    </h2>
    <div class=\\"accordion-collapse show\\">
      <div class=\\"accordion-body\\">
        <div class=\\"p-3 text-center\\">
          <div class=\\"input-range-track\\"
               style=\\"transform: scale(1); cursor: pointer;\\"
          >
            <div class=\\"input-range-thumb\\"
                 style=\\"position: absolute; z-index: 0; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);\\"
                 tabindex=\\"0\\"
                 aria-valuemax=\\"120\\"
                 aria-valuemin=\\"10\\"
                 aria-valuenow=\\"30\\"
                 draggable=\\"false\\"
                 role=\\"slider\\"
            >
              30s
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button \\"
              type=\\"button\\"
      >
        Filter bar configuration
      </button>
    </h2>
    <div class=\\"accordion-collapse show\\">
      <div class=\\"accordion-body\\">
        <div class=\\"form-check form-check-inline px-0 mx-0\\">
          <span class=\\"form-check form-switch\\">
            <input id=\\"configuration-autohide\\"
                   class=\\"form-check-input\\"
                   type=\\"checkbox\\"
                   checked
            >
            <label class=\\"form-check-label cursor-pointer me-3\\"
                   for=\\"configuration-autohide\\"
            >
              Hide filter bar and alert details when idle
            </label>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button \\"
              type=\\"button\\"
      >
        Theme
      </button>
    </h2>
    <div class=\\"accordion-collapse show\\">
      <div class=\\"accordion-body\\">
        <div class=\\"mb-2\\">
          <div class=\\" css-2b097c-container\\">
            <span aria-live=\\"polite\\"
                  aria-atomic=\\"false\\"
                  aria-relevant=\\"additions text\\"
                  class=\\"css-1f43avz-a11yText-A11yText\\"
            >
            </span>
            <div class=\\"react-select__control css-cat782-control\\">
              <div class=\\"react-select__value-container react-select__value-container--has-value css-17yk2dy-ValueContainer\\">
                <div class=\\"react-select__single-value css-1wh03ml-singleValue\\">
                  Automatic theme, follow browser preference
                </div>
                <div class=\\"css-ps6ina-Input\\">
                  <div class=\\"react-select__input\\"
                       style=\\"display: inline-block;\\"
                  >
                    <input autocapitalize=\\"none\\"
                           autocomplete=\\"off\\"
                           autocorrect=\\"off\\"
                           id=\\"react-select-configuration-theme-input\\"
                           spellcheck=\\"false\\"
                           tabindex=\\"0\\"
                           type=\\"text\\"
                           aria-autocomplete=\\"list\\"
                           style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                           value
                    >
                    <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
                    </div>
                  </div>
                </div>
              </div>
              <div class=\\"react-select__indicators css-vcwr3k-IndicatorsContainer\\">
                <span class=\\"react-select__indicator-separator css-1okebmr-indicatorSeparator\\">
                </span>
                <div class=\\"react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer\\"
                     aria-hidden=\\"true\\"
                >
                  <svg height=\\"20\\"
                       width=\\"20\\"
                       viewbox=\\"0 0 20 20\\"
                       aria-hidden=\\"true\\"
                       focusable=\\"false\\"
                       class=\\"css-tj5bde-Svg\\"
                  >
                    <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
                    </path>
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"mb-0\\">
          <div class=\\"form-check form-check-inline mx-0 px-0\\">
            <span class=\\"form-check form-switch\\">
              <input id=\\"configuration-colortitlebar\\"
                     class=\\"form-check-input\\"
                     type=\\"checkbox\\"
              >
              <label class=\\"form-check-label cursor-pointer me-3\\"
                     for=\\"configuration-colortitlebar\\"
              >
                Color group titlebar
              </label>
            </span>
          </div>
        </div>
        <div class=\\"mb-0\\">
          <div class=\\"form-check form-check-inline mx-0 px-0\\">
            <span class=\\"form-check form-switch\\">
              <input id=\\"configuration-animations\\"
                     class=\\"form-check-input\\"
                     type=\\"checkbox\\"
                     checked
              >
              <label class=\\"form-check-label cursor-pointer me-3\\"
                     for=\\"configuration-animations\\"
              >
                Enable animations
              </label>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button \\"
              type=\\"button\\"
      >
        Minimal alert group width
      </button>
    </h2>
    <div class=\\"accordion-collapse show\\">
      <div class=\\"accordion-body\\">
        <div class=\\"p-3 text-center\\">
          <div class=\\"input-range-track\\"
               style=\\"transform: scale(1); cursor: pointer;\\"
          >
            <div class=\\"input-range-thumb\\"
                 style=\\"position: absolute; z-index: 0; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);\\"
                 tabindex=\\"0\\"
                 aria-valuemax=\\"800\\"
                 aria-valuemin=\\"300\\"
                 aria-valuenow=\\"420\\"
                 draggable=\\"false\\"
                 role=\\"slider\\"
            >
              420
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button \\"
              type=\\"button\\"
      >
        Default number of alerts to show per group
      </button>
    </h2>
    <div class=\\"accordion-collapse show\\">
      <div class=\\"accordion-body\\">
        <div class=\\"p-3 text-center\\">
          <div class=\\"input-range-track\\"
               style=\\"transform: scale(1); cursor: pointer;\\"
          >
            <div class=\\"input-range-thumb\\"
                 style=\\"position: absolute; z-index: 0; cursor: grab; user-select: none; transform: translate(NaNpx, NaNpx);\\"
                 tabindex=\\"0\\"
                 aria-valuemax=\\"10\\"
                 aria-valuemin=\\"1\\"
                 aria-valuenow=\\"5\\"
                 draggable=\\"false\\"
                 role=\\"slider\\"
            >
              5
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button \\"
              type=\\"button\\"
      >
        Default alert group display
      </button>
    </h2>
    <div class=\\"accordion-collapse show\\">
      <div class=\\"accordion-body\\">
        <div class=\\"mb-0\\">
          <div class=\\" css-2b097c-container\\">
            <span aria-live=\\"polite\\"
                  aria-atomic=\\"false\\"
                  aria-relevant=\\"additions text\\"
                  class=\\"css-1f43avz-a11yText-A11yText\\"
            >
            </span>
            <div class=\\"react-select__control css-cat782-control\\">
              <div class=\\"react-select__value-container react-select__value-container--has-value css-17yk2dy-ValueContainer\\">
                <div class=\\"react-select__single-value css-1wh03ml-singleValue\\">
                  Collapse on mobile
                </div>
                <div class=\\"css-ps6ina-Input\\">
                  <div class=\\"react-select__input\\"
                       style=\\"display: inline-block;\\"
                  >
                    <input autocapitalize=\\"none\\"
                           autocomplete=\\"off\\"
                           autocorrect=\\"off\\"
                           id=\\"react-select-configuration-collapse-input\\"
                           spellcheck=\\"false\\"
                           tabindex=\\"0\\"
                           type=\\"text\\"
                           aria-autocomplete=\\"list\\"
                           style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                           value
                    >
                    <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
                    </div>
                  </div>
                </div>
              </div>
              <div class=\\"react-select__indicators css-vcwr3k-IndicatorsContainer\\">
                <span class=\\"react-select__indicator-separator css-1okebmr-indicatorSeparator\\">
                </span>
                <div class=\\"react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer\\"
                     aria-hidden=\\"true\\"
                >
                  <svg height=\\"20\\"
                       width=\\"20\\"
                       viewbox=\\"0 0 20 20\\"
                       aria-hidden=\\"true\\"
                       focusable=\\"false\\"
                       class=\\"css-tj5bde-Svg\\"
                  >
                    <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
                    </path>
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button \\"
              type=\\"button\\"
      >
        Grid sort order
      </button>
    </h2>
    <div class=\\"accordion-collapse show\\">
      <div class=\\"accordion-body\\">
        <div class=\\"mb-0\\">
          <div class=\\"d-flex flex-fill flex-lg-row flex-column justify-content-between\\">
            <div class=\\"flex-shrink-0 flex-grow-1 flex-basis-auto\\">
              <div class=\\" css-2b097c-container\\">
                <span aria-live=\\"polite\\"
                      aria-atomic=\\"false\\"
                      aria-relevant=\\"additions text\\"
                      class=\\"css-1f43avz-a11yText-A11yText\\"
                >
                </span>
                <div class=\\"react-select__control css-cat782-control\\">
                  <div class=\\"react-select__value-container react-select__value-container--has-value css-17yk2dy-ValueContainer\\">
                    <div class=\\"react-select__single-value css-1wh03ml-singleValue\\">
                      Use defaults from karma config file
                    </div>
                    <div class=\\"css-ps6ina-Input\\">
                      <div class=\\"react-select__input\\"
                           style=\\"display: inline-block;\\"
                      >
                        <input autocapitalize=\\"none\\"
                               autocomplete=\\"off\\"
                               autocorrect=\\"off\\"
                               id=\\"react-select-configuration-sort-order-input\\"
                               spellcheck=\\"false\\"
                               tabindex=\\"0\\"
                               type=\\"text\\"
                               aria-autocomplete=\\"list\\"
                               style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                               value
                        >
                        <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
                        </div>
                      </div>
                    </div>
                  </div>
                  <div class=\\"react-select__indicators css-vcwr3k-IndicatorsContainer\\">
                    <span class=\\"react-select__indicator-separator css-1okebmr-indicatorSeparator\\">
                    </span>
                    <div class=\\"react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer\\"
                         aria-hidden=\\"true\\"
                    >
                      <svg height=\\"20\\"
                           width=\\"20\\"
                           viewbox=\\"0 0 20 20\\"
                           aria-hidden=\\"true\\"
                           focusable=\\"false\\"
                           class=\\"css-tj5bde-Svg\\"
                      >
                        <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
                        </path>
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"accordion-item\\">
    <h2 class=\\"accordion-header\\">
      <button class=\\"accordion-button \\"
              type=\\"button\\"
      >
        Multi-grid source label
      </button>
    </h2>
    <div class=\\"accordion-collapse show\\">
      <div class=\\"accordion-body\\">
        <div class=\\"mb-0\\">
          <div class=\\"d-flex flex-fill flex-lg-row flex-column justify-content-between\\">
            <div class=\\"flex-shrink-0 flex-grow-1 flex-basis-auto mx-0 mx-lg-1 mt-1 mt-lg-0\\">
              <div class=\\" css-2b097c-container\\">
                <span aria-live=\\"polite\\"
                      aria-atomic=\\"false\\"
                      aria-relevant=\\"additions text\\"
                      class=\\"css-1f43avz-a11yText-A11yText\\"
                >
                </span>
                <div class=\\"react-select__control css-cat782-control\\">
                  <div class=\\"react-select__value-container react-select__value-container--has-value css-17yk2dy-ValueContainer\\">
                    <div class=\\"react-select__single-value css-1wh03ml-singleValue\\">
                      Disable multi-grid
                    </div>
                    <div class=\\"css-ps6ina-Input\\">
                      <div class=\\"react-select__input\\"
                           style=\\"display: inline-block;\\"
                      >
                        <input autocapitalize=\\"none\\"
                               autocomplete=\\"off\\"
                               autocorrect=\\"off\\"
                               id=\\"react-select-configuration-grid-label-input\\"
                               spellcheck=\\"false\\"
                               tabindex=\\"0\\"
                               type=\\"text\\"
                               aria-autocomplete=\\"list\\"
                               style=\\"box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;\\"
                               value
                        >
                        <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
                        </div>
                      </div>
                    </div>
                  </div>
                  <div class=\\"react-select__indicators css-vcwr3k-IndicatorsContainer\\">
                    <span class=\\"react-select__indicator-separator css-1okebmr-indicatorSeparator\\">
                    </span>
                    <div class=\\"react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer\\"
                         aria-hidden=\\"true\\"
                    >
                      <svg height=\\"20\\"
                           width=\\"20\\"
                           viewbox=\\"0 0 20 20\\"
                           aria-hidden=\\"true\\"
                           focusable=\\"false\\"
                           class=\\"css-tj5bde-Svg\\"
                      >
                        <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
                        </path>
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div class=\\"flex-shrink-1 flex-grow-0 form-check form-check-inline flex-basis-auto my-1 my-lg-auto ms-0 ms-lg-2 me-0 px-0\\">
              <span class=\\"form-check form-switch\\">
                <input id=\\"configuration-multigrid-sort-reverse\\"
                       class=\\"form-check-input\\"
                       type=\\"checkbox\\"
                >
                <label class=\\"form-check-label cursor-pointer\\"
                       for=\\"configuration-multigrid-sort-reverse\\"
                >
                  Reverse order
                </label>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
"
`;
